internal/ingress/controller/template/template.go

```diff
@@ -287,9 +288,26 @@ func buildResolvers(res interface{}, disableIpv6 interface{}) string {
 	return strings.Join(r, " ") + ";"
 }
 
+// ususRewriteAnnotation checks if any location uses the nginx.ingress.kubernetes.io/rewrite-target annotation
+func usesRewriteAnnotation(input interface{}) bool {
+	locations, ok := input.([]*ingress.Location)
+	if !ok {
+		glog.Errorf("expected an '[]*ingress.Location' type but %T was returned", input)
+		return false
+	}
+
+	for _, location := range locations {
+		path := location.Path
+		if len(location.Rewrite.Target) > 0 && location.Rewrite.Target != path {
```

Comment:
Can you extract this and https://github.com/kubernetes/ingress-nginx/pull/3078/files#diff-652deabba49a2a87d4c5a79700dfa1e3R318 into a separate utility function, i.e `needsRewrite(location)`? Since they are doing the same thing, and extracting it out will also increase readability of the code.